Sources/Error.swift

```diff
@@ -2,22 +2,26 @@ public enum Error: ErrorType, Equatable {
     case General
     case InvalidUrl(String)
     case InvalidResponseData(NSData?)
-    case Error(NSError?)
+    case ErrorNS(NSError?)
+    case Error(ErrorType)
     case InvalidAuthentication
     case ShouldOverride
+    case CreateDataTask
```

Comment:
But since you create the task before calling the `if-let` clause, this will never be nil. You can use `task?.resume()`, no?